core/src/wallet/bitcoin/BitcoinLikeAccount.cpp

```diff
@@ -531,14 +531,32 @@ namespace ledger {
             });
         }
 
+        // This method is used to get a block height only used to know which
+        // update/fork (activated at certain block height) is used.
+        // This explains why we take LLONG_MAX is we have no block in DB for
+        // certain currency
+        // WARNING: please don't use this method if you need an accurate value
+        // of last block
+        static uint64_t getLastBlockFromDB(soci::session &sql, const std::string &currencyName) {
+            //Get last block from DB
+            auto lastBlock= BlockDatabaseHelper::getLastBlock(sql, currencyName);
+            //If we can not retrieve a last block for currency, we apply last BIP/update/fork
+            return lastBlock.hasValue() ? static_cast<uint64_t >(lastBlock.getValue().height) : LLONG_MAX;
```

Comment:
Does it mean we retreive a whole block just for discarding it and use only the height? I’m fine with this as first implementation but I guess it could be more efficient to have a way to get only the height (without getting payload) from the `BlockDatabaseHelper`. That means adding `BlockDatabaseHelper::getLastBlockHeight` or something like that.

Reply:
Also, I _think_ you can change this:

```
lastBlock.getValue().height
```

with this:

```
lastBlock->heigth
```

(there’s a prefix `operator*` for `Optional<_>` but I’m not sure there is for `operator->`)

Reply:
For the payload issue, I agree on the concept, here the idea was just reusing an existing method, more generally we should have a more generic method to interact/retrieve data with/from DB but it should be easier with the new DB abstraction